components/MySQLConnection.php

```diff
@@ -47,4 +49,28 @@ public function getConnection():mysqli {
             return self::$connection;
         }
 
+        /**
+         * Returns the result of MySQLi query
+         *
+         * @param Query $query
+         * @return mysqli_result|bool
+         */
+        public function query(Query $query) {
+            return self::$connection->query($query->getQueryString());
+        }
```

Comment:
Encapsulates the extraction of the query string which means that it cannot be substituted mid-process.